q2_longitudinal/_utilities.py

```diff
@@ -728,14 +729,36 @@ def _first_differences(metadata, state_column, individual_id_column, metric,
     pairs_summary = pd.DataFrame()
     errors = []
     states = sorted(metadata[state_column].unique())
+
+    # if calculating static differences, validate baseline as a valid state
+    if baseline is not None:
```

Comment:
```bash
$ qiime longitudinal first-differences \
  --m-metadata-file ecam-sample-metadata.tsv \
  --m-metadata-file shannon.qza \
  --p-state-column month \
  --p-metric shannon \
  --o-first-differences shannon-first-differences.qza \
  --p-individual-id-column studyid \
  --p-replicate-handling random \
  --p-baseline foo \
  --verbose
Traceback (most recent call last):
  File "/home/matthew/.conda/envs/q2dev-2017.11/lib/python3.5/site-packages/q2cli/commands.py", line 218, in __call__
    results = action(**arguments)
  File "<decorator-gen-350>", line 2, in first_differences
  File "/home/matthew/.conda/envs/q2dev-2017.11/lib/python3.5/site-packages/qiime2/sdk/action.py", line 220, in bound_callable
    output_types, provenance)
  File "/home/matthew/.conda/envs/q2dev-2017.11/lib/python3.5/site-packages/qiime2/sdk/action.py", line 355, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/home/matthew/src/qiime2/q2-longitudinal/q2_longitudinal/_longitudinal.py", line 229, in first_differences
    replicate_handling, baseline=baseline, distance_matrix=None)
  File "/home/matthew/src/qiime2/q2-longitudinal/q2_longitudinal/_utilities.py", line 736, in _first_differences
    baseline = metadata[state_column].dtype.type(baseline)
ValueError: invalid literal for int() with base 10: 'foo'

Plugin error from longitudinal:

  invalid literal for int() with base 10: 'foo'

See above for debug info.
```

This is related to the discussion below about this parameter's type (`Str` vs some kind of numerical type). If the right type was declared in the method registration, this kind of error would be prevented by the framework.